127-cd-nextgen-entities/src/main/java/io/harness/cdng/ssh/SshWinRmConfigFileHelper.java

```
@@ -83,8 +90,16 @@ public FileDelegateConfig getFileDelegateConfig(
     List<StoreDelegateConfig> stores = new ArrayList<>(configFilesOutcome.size());
     for (ConfigFileOutcome configFileOutcome : configFilesOutcome.values()) {
       StoreConfig storeConfig = configFileOutcome.getStore();
-      if (storeConfig != null && HARNESS_STORE_TYPE.equals(storeConfig.getKind())) {
-        stores.add(buildHarnessStoreDelegateConfig(ambiance, (HarnessStore) storeConfig, shouldRenderConfigFiles));
+      if (storeConfig != null) {
+        switch (storeConfig.getKind()) {
+          case HARNESS_STORE_TYPE:
+            stores.add(buildHarnessStoreDelegateConfig(ambiance, (HarnessStore) storeConfig, shouldRenderConfigFiles));
+            break;
+          case GITHUB:
+            stores.add(buildGitFetchedStoreDelegateConfig(ambiance, configFileOutcome));
+            break;
+          default: // do nothing
```

we should throw the unsupported exception for ones that we are not supporting